nearcore/src/state_sync.rs

```diff
@@ -202,26 +202,50 @@ fn select_random_part_id_with_index(parts_to_be_dumped: &Vec<u64>) -> (u64, usiz
     (selected_element, selected_idx)
 }
 
+struct StateDumpShardInfo {
+    epoch_id: EpochId,
+    epoch_height: EpochHeight,
+    sync_hash: CryptoHash,
+}
+
 fn get_current_state(
     chain: &Chain,
     shard_id: &ShardId,
     shard_tracker: &ShardTracker,
     account_id: &Option<AccountId>,
     epoch_manager: Arc<dyn EpochManagerAdapter>,
-) -> Result<Option<(EpochId, EpochHeight, CryptoHash)>, Error> {
-    let was_last_epoch_dumped = match chain.chain_store().get_state_sync_dump_progress(*shard_id) {
+) -> Result<Option<StateDumpShardInfo>, Error> {
```

Comment:
I was actually suggesting to remove the option and instead have a new enum for this. But ff you can just explain what does None return value mean in a comment that's good enough for me too :)

Reply:
Changed the result to `Dump{...}` or `Wait`. This should explain it.